packages/react-components/src/components/VideoGallery/VideoGallery.tsx

```diff
@@ -105,6 +114,49 @@ export const VideoGallery = (props: VideoGalleryProps): JSX.Element => {
     return !!(layout === 'floatingLocalVideo' && remoteParticipants && remoteParticipants.length > 0);
   }, [layout, remoteParticipants]);
 
+  const containerRef = useRef<HTMLDivElement>(null);
+  const isMobileScreen = useIsSmallScreen(containerRef);
+  const visibleVideoParticipants = useRef<VideoGalleryParticipant[] | []>([]);
+  const visibleAudioParticipants = useRef<VideoGalleryParticipant[] | []>([]);
+  const [videoParticipants, setVideoParticipants] = useState<VideoGalleryParticipant[] | []>();
+  const [audioParticipants, setAudioParticipants] = useState<VideoGalleryParticipant[] | []>();
+
+  useEffect(() => {
+    visibleVideoParticipants.current = smartDominantSpeakerParticipants(
+      remoteParticipants?.filter((p) => p.videoStream?.isAvailable) ?? [],
+      dominantSpeakers,
+      visibleVideoParticipants.current.filter((p) => p.videoStream?.isAvailable)
+    );
+
+    // Create a map of visibleVideoParticipants for faster searching.
+    // This map will be used to identify overflow participants. i.e., participants
+    // that should be rendered in horizontal gallery.
+    const visibleVideoParticipantsMap = {};
+    visibleVideoParticipants.current.forEach((p) => {
+      visibleVideoParticipantsMap[p.userId] = true;
+    });
+    // @TODO: Can this possibly be done inside HorizontalGallery?
```

Comment:
Resolve todo unless there's a realistic goal that someone will look at this in the future